### PR TITLE
FE Architecture change for Dough Core - Story TP-9894 (Task: TP-9896)

### PR DESCRIPTION
**Pull request by nmiddleweek**
This PR is part one of three in completing the FEA merge into master.

This relates to TP-9896 and is part of the Story TP-9894, see following: 

* https://moneyadviceservice.tpondemand.com/entity/9896-merge-fe-architecture-into-master-for
* https://moneyadviceservice.tpondemand.com/entity/9894-pacs-merge-fe-architecure-branches-into

### CODE REVIEW

**Comment by masbuild** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">It looks like this is a Rails engine, but no changes to <a href="https://github.com/moneyadviceservice/dough/blob/08d6f38ce1d52614304f7445b3fbeaea69a00115/lib/dough/version.rb">lib/dough/version.rb</a> detected. Did you forget to bump the version?</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">English locale contains keys missing in Welsh locale. Missing keys: [“taken”, “taken”].</td>
    </tr>
  </tbody>
</table>

At Money Advice Service, we use Danger gem, to ensure that our commits and pull requests follow [standards](https://github.com/moneyadviceservice/mas-standards/blob/master/Dangerfile). If you think one of the points here has been raised in error (or shouldn't be in our standards), open an issue or pull request on [our Danger config](https://github.com/moneyadviceservice/mas-standards).

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: spec/js/tests/DoughBaseComponent_spec.js [hunk 1]
```diff
@@ -92,6 +98,26 @@ describe('DoughBaseComponent', function() {
...
       });
+
+      it('should stamp an id attribute on the component element', function() {
+        var doughBaseComponent = new this.DoughBaseComponent(this.component);
+
+        doughBaseComponent._initialisedSuccess(initialised);
+
+        expect(doughBaseComponent.$el.attr('data-dough-dough-base-component-id'))
+          .to
+          .match(/[0-9]+/);
```
Comment (davidtrussler) [inline spec/js/tests/DoughBaseComponent_spec.js]: No big deal but maybe make lose the line breaks here for consistency.

File: spec/js/tests/DoughBaseComponent_spec.js [hunk 2]
```diff
@@ -92,6 +98,26 @@ describe('DoughBaseComponent', function() {
...
+      });
+
+      it('should trigger a successful event upon initialisation', function() {
+        var doughBaseComponent = new this.DoughBaseComponent(this.component),
+            spy = sandbox.spy();
+
+        doughBaseComponent.$el.on('INITIALISE-SUCCESS.DoughBaseEvent', spy);
+        doughBaseComponent._initialisedSuccess(initialised);
+
+        expect(spy.called).to.be.true;
```
Comment (davidtrussler) [inline spec/js/tests/DoughBaseComponent_spec.js]: maybe better to use calledOnce here

File: spec/js/tests/DoughBaseComponent_spec.js (hunk 1 above)
Comment (nmiddleweek) [reply spec/js/tests/DoughBaseComponent_spec.js]: Must admit, @willhall88 did that. I quite like it myself for ease of reading. I'll wrap it from 3 to 2 lines.

File: spec/js/tests/DoughBaseComponent_spec.js [hunk 3]
```diff
@@ -92,6 +98,26 @@ describe('DoughBaseComponent', function() {
...
+        expect(doughBaseComponent.$el.attr('data-dough-dough-base-component-id'))
+          .to
+          .match(/[0-9]+/);
+      });
+
+      it('should trigger a successful event upon initialisation', function() {
+        var doughBaseComponent = new this.DoughBaseComponent(this.component),
+            spy = sandbox.spy();
+
+        doughBaseComponent.$el.on('INITIALISE-SUCCESS.DoughBaseEvent', spy);
```
Comment (nmiddleweek) [inline spec/js/tests/DoughBaseComponent_spec.js]: I think we should be using the DoughEventConstants constants here so I'll update.

File: spec/js/tests/DoughBaseComponent_spec.js (hunk 2 above)
Comment (nmiddleweek) [reply spec/js/tests/DoughBaseComponent_spec.js]: Look like Will followed the convention that was already in the spec file, see line 192. I'll leave this one as-is.

File: spec/js/tests/DoughBaseComponent_spec.js (hunk 1 above)
Comment (willhall88) [reply spec/js/tests/DoughBaseComponent_spec.js]: This is from my ruby days probably. Breaking long lines onto multiple and the convention is to then line break at functions

File: spec/js/tests/DoughBaseComponent_spec.js (hunk 2 above)
Comment (willhall88) [reply spec/js/tests/DoughBaseComponent_spec.js]: `calledOnce` is probably a clearer and more specific test. I'm happy for it to be changed

**Pull request merged**